### DISCUSSION TITLE
keyboard interactions

### DISCUSSION THREAD

**Discussion by karussell** [openlayers/openlayers#15406] [Q&A] [answered]
I think the keyboard interactions are enabled by default, but they don't work for [our application](https://graphhopper.com/maps/) ([code](https://github.com/graphhopper/graphhopper-maps/blob/2340ebbc69d4fc07759cad7d08f8884dade997bf/src/map/map.ts#L10)). 

Probably this is related to the following snippet in the documentation (which I unfortunately do not understand):

> By default, this is the map div, though you can change this with the keyboardEventTarget in [Map](https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html). document never loses focus but, for any other element, focus will have to be on, and returned to, this element if the keys are to function

I also tried to explicitly add these interactions and enable `keyboard: true` but without success.

Furthermore I did not find an example where this works (in Firefox) even after clicking on the map to focus it.

Can someone give me a hint :) ?

**Comment by ahocevar** [comment, marked as answer]
Just add `tabindex=“0”` to the target element of your map.

**Reply by karussell** [reply]
Unfortunately now I need to click [on the map with a route](https://graphhopper.com/maps/?point=51.049329%2C13.738144_Dresden%2C+Saxony%2C+Germany&point=48.137108%2C11.575382_Munich%2C+Bavaria%2C+Germany&profile=car&layer=Omniscale) before I can zoom with the mouse wheel (or touch gesture).

The [main page](https://graphhopper.com/maps/) without a route works.

Could it be related to [another location where we use tabindex](https://github.com/graphhopper/graphhopper-maps/blob/5053f0f61e61c1ca29400458fcf2a9207548fa14/src/sidebar/search/Search.tsx#L113) too?

If this is a generic JS thing and unrelated to openlayers please let me know and I'll have to study this tabindex further and elsewhere :)

**Reply by ahocevar** [reply]
Yes, it's a generic web development thing. The number you pass to `tabindex` defines the order of elements you want to focus, and you can programmatically use `window.focus(element)` to focus a specific element.

**Comment by karussell** [comment]
Oh, cool. It works - thanks a lot!

(Why is that? And how did you know this :D ? )

**Comment by ahocevar** [comment]
It‘s documented with the `keyboardEventTarget` option: https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html
